<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <link href='https://fonts.googleapis.com/css?family=Architects+Daughter' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">

    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <title>ASF Characterization</title>
  </head>

  <body>
    <header>
      <div class="inner">
        <h2>ASF Characterization</h2>
        <h3>Bioinformatics, Metabolism, Interactions of the Altered Schaedler Flora (ASF)</h3>
        <a href="https://github.com/mbi2gs/asf_characterization" class="button"><small>View project on</small> GitHub</a>
      </div>
    </header>

<div id="content-wrapper">
	<div class="inner clearfix">
	
<section id="main-content">
<h2>Overview</h2>
<p>Welcome to the code guide for this <a href="https://github.com/mbi2gs/asf_characterization/" target="_blank">project repository</a>, 
containing data and associated analysis scripts related 
to the bioinformatic and <em>in vitro</em> characterization of the Altered 
Schaedler Flora (ASF).</p>

<h2>Comparative Bioinformatic Analysis of Gene Functional Categories</h2>
<p>All 
<a href="https://github.com/mbi2gs/asf_characterization/tree/master/AnnotationAndMetagenomics/bin" target="_blank">code</a>, 
<a href="https://github.com/mbi2gs/asf_characterization/tree/master/AnnotationAndMetagenomics/data/phylum_nogs" target="_blank">data</a>, and 
<a href="https://github.com/mbi2gs/asf_characterization/tree/master/AnnotationAndMetagenomics/results" target="_blank">results</a>
files relating to the bioinformatics comparison of 15 wild murine metagenomes with the ASF composite metagenome. The
<a href="https://github.com/mbi2gs/asf_characterization/blob/master/AnnotationAndMetagenomics/README.md" target="_blank">README</a>
file contains specific information about the contents.
</p>

<h2>Spent Media Experiments</h2>
<h3>Defined Growth Media</h3>
<p>Instructions to make 
	<a href="https://github.com/mbi2gs/asf_characterization/blob/master/SupplementedLBMedia/Supplemented_LB_media_for_ASF_studies.docx" target="_blank">
Supplemented LB</a> which supports growth of 7/8 ASF members.</p>

<h3>Growth Curves</h3>
<p>Growth curves files (optical density) are available in the folder 
	<a href="https://github.com/mbi2gs/asf_characterization/tree/master/GrowthCurves/SpentMediaGrowthCurveData" target="_blank">
"GrowthCurves/SpentMediaGrowthCurveData/"</a>.</p>
<p>The R script 
	<a href="https://github.com/mbi2gs/asf_characterization/blob/master/Metabolomics/plot_relative_intensities.R" target="_blank">
"plot_growth_curves.R"</a> reads in the growth curve files and organizes them into a single plot ("ASF_SpentMedia_GrowthCurves.tiff").
It further calculates the area under each curve and outputs those into a tab-separated file ("auc_inhibitions_all_pairs.tsv").</p>

<img src="images/ASF519in360.jpg" alt="Growth Curves" style="width:304px;height:228px;"></br>

<h3>Metabolomics</h3>
<p>NMR peak integral files and sample metadata are available in the folder 
	<a href="https://github.com/mbi2gs/asf_characterization/tree/master/Metabolomics/NMR_Integrals" target="_blank">
"Metabolomics/NMR_Integrals/"</a>.</p>
<p>The R script 
	<a href="https://github.com/mbi2gs/asf_characterization/blob/master/Metabolomics/metabolomics_analysis_and_plotting.R" target="_blank">
"metabolomics_analysis_and_plotting.R"</a> reads in the peak integral file and calculates z-scores for each sample with respect to fresh media.
This script plots the first round of data and the second round (ASF492 re-do samples) in ("ASF_metabolite_zscores_plots.pdf") which
allows for comparisons between the two batches of data. The R script further plots the z-scores as heat maps, counts the 
different categories of metabolite changes between conditions, and identifies emergent metabolic phenotypes.</p>

<img src="images/519in360 .jpg" alt="Metabolite Z-scores" style="width:304px;height:228px;"></br>

</section>

	<aside id="sidebar">
	  <a href="https://github.com/mbi2gs/asf_characterization/zipball/master" class="button">
		<small>Download</small>
		.zip file
	  </a>
	  <a href="https://github.com/mbi2gs/asf_characterization/tarball/master" class="button">
		<small>Download</small>
		.tar.gz file
	  </a>

	  <p class="repo-owner"><a href="https://github.com/mbi2gs/asf_characterization"></a> is maintained by <a href="https://github.com/mbi2gs">mbi2gs</a>.</p>

	  <p>This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the Architect theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.</p>
	</aside>
	
  </div>
</div>

  
  </body>
</html>
